app/graphql/resolvers/submission_resolver.rb

```diff
@@ -6,7 +6,9 @@ class SubmissionResolver < BaseResolver
   def run
     check_submission_presence!
 
-    unless draft_in_progress?(submission, @arguments) || admin? || partner?
+    unless draft_in_progress?(submission, @arguments) ||
+             matching_email(submission, arguments&.[](:user_email)) || admin? ||
```

Comment:
Is `arguments&.[](:user_email)` working as expected?
I would change it to `@arguments&.[](:user_email)` since we are using it as `@arguments` in the first condition `draft_in_progress?(submission, @arguments)`.